fuzzing/fuzztools.py

```diff
@@ -49,52 +43,51 @@
 # OPENERS = [AAC]
 
 
-def run(opener, data):
-    f = BytesIO(data)
+def run(opener, f):
     try:
         res = opener(f)
     except MutagenError:
-        pass
-    else:
-        # File is special and returns None if loading fails
-        if opener is File and res is None:
-            return
+        return
 
-        # These can still fail because we might need to parse more data
-        # to rewrite the file
+    # File is special and returns None if loading fails
+    if opener is File and res is None:
+        return
 
-        f.seek(0)
-        try:
-            res.save(f)
-        except MutagenError:
-            pass
+    # These can still fail because we might need to parse more data
+    # to rewrite the file
 
-        f.seek(0)
-        res = opener(f)
+    f.seek(0)
+    try:
+        res.save(f)
+    except MutagenError:
+        return
```

Comment:
Might we miss issues because of this early return? Same below for the `delete` calls